osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/monitoring/GetPowerQualityProfileCommandExecutorTest.java

```diff
@@ -66,19 +67,66 @@ public void testExecute() throws ProtocolAdapterException {
         when(dlmsHelper.readLong(any(DataObject.class), any(String.class))).thenCallRealMethod();

```

Comment:
Ik vind het moeilijk te beoordelen of deze test volledig is. Wat ik wel kan beoordelen is dat ik uit deze test niet kan opmaken wat de implementatie zou moeten doen. Dit kun je verbeteren door duidelijk test methode namen en laten zien wat je als inhoud verwacht. Tenzij de size van de verschillende collecties dat is waar het om gaat, maar wellicht is de inhoud van de collecties ook belangrijk?

Reply:
er zit heel veel logica in de dlmshelper, hier is een aparte test voor. deze test transformeert data van de meter (die ik inschiet via de mock) naar een DTO. de ene handler filtert vooraf, en de andere achteraf.

Reply:
De unit testen komen niet 1 op 1 overeen met de classes. Dan willen we wel, lijkt me?

Reply:
nee het idee was, omdat de executor alleen maar een if-then-else statement is, om deze testen samen te voegen tot 1 test, beetje lui maar wel effectief :-)

ik ben deze nu aan het herschikken naar 2 handler tests.

Reply:
I don't know how everything should work together, but when I run the test with coverage GetPowerQualityProfileCommandExecutor.execute() isn't covered; this is weird because it is the only method in the class under test.

Reply:
Oh, sorry, duplicate apparently :-)

Reply:
@jandik I reshuffled the tests